comp/core/autodiscovery/stream/stream.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package stream provides the autodiscovery stream config
+package stream
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/proto"
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
+	"github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Config streams autodiscovery configs
+func Config(ac autodiscovery.Component, out pb.AgentSecure_AutodiscoveryStreamConfigServer) error {
+	s := &scheduler{
+		out:  out,
+		done: make(chan error, 1),
+	}
+
+	// TODO: add a uuid to avoid name collision when there are concurrent rpc calls ?
+	schedulerName := "remote"
+	ac.AddScheduler(schedulerName, s, true)
+	defer ac.RemoveScheduler(schedulerName)
+
+	return <-s.done
+}
+
+type scheduler struct {
+	out  pb.AgentSecure_AutodiscoveryStreamConfigServer
+	done chan error
+}
+
+func (s *scheduler) Schedule(config []integration.Config) {
+	s.handleEvent(config, pb.ConfigEventType_SCHEDULE)
+}
+
+func (s *scheduler) Unschedule(configs []integration.Config) {
+	s.handleEvent(configs, pb.ConfigEventType_UNSCHEDULE)
+}
+
+func (s *scheduler) Stop() {
+	close(s.done)
+}
+
+func (s *scheduler) handleEvent(configs []integration.Config, eventType pb.ConfigEventType) {
+	protobufConfigs := protobufConfigFromAutodiscoveryConfigs(configs, eventType)
+
+	err := grpc.DoWithTimeout(func() error {
+		return s.out.Send(&pb.AutodiscoveryStreamResponse{
+			Configs: protobufConfigs,
+		})
+	}, 1*time.Minute)
+
+	if err != nil {
+		log.Warnf("error sending %s autodiscovery event: %s", eventType.String(), err)
+		s.done <- err
+	}
```

Comment:
Should we exit if there is an error when sending an event? If we disconnect here, we need to figure a way reconnect. 

What other option do you think we could do?

Reply:
I kept that logic from the AK branch, but I'm wondering the same... 🤔

Reply:
We could implement a retry mechanism, but that's already what happens inside `Send` (I think ?) so I'm not sure it's super useful